<a href="https://colab.research.google.com/github/LuizBecker01/PensamentoComputacional/blob/main/Lista_de_exerc%C3%ADcios_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Terceira lista de exercicios

###1. Faça o teste de mesa no algoritmo a seguir:


``` python
  ALGORITMO TABUADA_9;
  VAR
    N, X, R : INTEIRO;
  INICIO
    PARA X DE 1 ATE 10 FACA
      INICIO
        R ← 9 * X;
        ESCREVER ( X, ‘ x 9 = ’, R);
      FIM;
    FIM_PARA;
  FIM.
```

In [ ]:
#tabuada do 9

x = int(input("Digite um numero:"))

for i in range (11):
  r = x * i
  print(i, "x 9 =", r)



Digite um numero:10
0 x 9 = 0
1 x 9 = 10
2 x 9 = 20
3 x 9 = 30
4 x 9 = 40
5 x 9 = 50
6 x 9 = 60
7 x 9 = 70
8 x 9 = 80
9 x 9 = 90
10 x 9 = 100


###2. Faça um programa em Python que mostre todos os números inteiros ímpares situados na faixa de 1000 a 1500.

In [ ]:
# Mostra os números ímpares de 1000 a 1500
for numero in range(1000, 1500):
  if numero % 2 != 0:
    print(numero)

###3. Elabore um programa em Python que calcule e escreva a soma de 10 números lidos.

In [ ]:
soma = 0

for i in range (10):
  numero = float(input(f"Digite {i+1} numero"))
  soma += numero

print(f"\nA soma dos 10 números é: {soma}")


Digite 1 numero10
Digite 2 numero15
Digite 3 numero20
Digite 4 numero30
Digite 5 numero40
Digite 6 numero50
Digite 7 numero60
Digite 8 numero70
Digite 9 numero80
Digite 10 numero90

A soma dos 10 números é: 465.0


###4. Faça um programa em Python que leia a nota de 10 alunos de uma turma. Ao final, deve ser escrita a média geral da turma.

In [ ]:
numero = 0

# faz com que adicione 10 notas digitadas pelo usuário somando elas
for i in range (10):
  numero += float(input(f"Digite {i+1} a sua nota"))

#calcula a média
media = numero / 10

print(media)
print(numero)

Digite 1 a sua nota5
Digite 2 a sua nota5
Digite 3 a sua nota5
Digite 4 a sua nota5
Digite 5 a sua nota5
Digite 6 a sua nota5
Digite 7 a sua nota5
Digite 8 a sua nota5
Digite 9 a sua nota5
Digite 10 a sua nota5
5.0
50.0


###5. Faça um programa em Python que leia 5 números. Para cada número fornecido, escreva se ele é NULO, NEGATIVO ou POSITIVO.

In [ ]:
x = 0

for i in range (5):
  x = int(input(f"Digite {i+1} numero " ))
  if x > 0:
    print("Positivo")
  elif x == 0:
    print("Nulo")
  else:
    print("Negativo")

Digite 1 numero 5
Positivo
Digite 2 numero 10
Positivo
Digite 3 numero -3
Negativo
Digite 4 numero 0
Nulo
Digite 5 numero 1
Positivo


###6. Ler dois valores inteiros e escrever os números em sequência do menor até maior valor.

In [ ]:
num1 = int(input("Digite um numero: "))
num2 = int(input("Digite o seundo numero"))

if num1 > num2:
  for i in range(num2, num1 + 1):
    print(i)
else:
  for i in range(num1, num2 + 1):
    print(i)

Digite um numero: 0
Digite o seundo numero5
0
1
2
3
4
5


###7. Faça um programa em Python que escreva os números pares entre 100 e 200.

In [ ]:
for numero in range(100, 200):
  if numero % 2 == 0:
    print(numero)

100
102
104
106
108
110
112
114
116
118
120
122
124
126
128
130
132
134
136
138
140
142
144
146
148
150
152
154
156
158
160
162
164
166
168
170
172
174
176
178
180
182
184
186
188
190
192
194
196
198


In [ ]:
for numero in range(100, 200, 2):
    print(numero)

100
102
104
106
108
110
112
114
116
118
120
122
124
126
128
130
132
134
136
138
140
142
144
146
148
150
152
154
156
158
160
162
164
166
168
170
172
174
176
178
180
182
184
186
188
190
192
194
196
198


###8. Escrever um programa em Python que leia um valor para n, inteiro e positivo, e escreva atabuada de n. Com os respectivos valores.

n x 1 =

n x 2 =

...

n x 10 =


In [ ]:
x = int(input("Digite um numero para multiplicar: "))

for i in range (11):
  r = x * i
  print(i, "x", x, "=", r)

Digite um numero para multiplicar: 10
0 x 10 = 0
1 x 10 = 10
2 x 10 = 20
3 x 10 = 30
4 x 10 = 40
5 x 10 = 50
6 x 10 = 60
7 x 10 = 70
8 x 10 = 80
9 x 10 = 90
10 x 10 = 100


###9. Escrever um programa em Python que gera os números de 1000 a 1999 e escreve aqueles que divididos por 11 dão um resto igual a 5.

In [ ]:
for i in range(1000, 1999):
  if i % 11 == 0:
    print(i)

1001
1012
1023
1034
1045
1056
1067
1078
1089
1100
1111
1122
1133
1144
1155
1166
1177
1188
1199
1210
1221
1232
1243
1254
1265
1276
1287
1298
1309
1320
1331
1342
1353
1364
1375
1386
1397
1408
1419
1430
1441
1452
1463
1474
1485
1496
1507
1518
1529
1540
1551
1562
1573
1584
1595
1606
1617
1628
1639
1650
1661
1672
1683
1694
1705
1716
1727
1738
1749
1760
1771
1782
1793
1804
1815
1826
1837
1848
1859
1870
1881
1892
1903
1914
1925
1936
1947
1958
1969
1980
1991


###10. Faça um programa em Python que leia um número qualquer (N), e realize a soma dos números de 1 até N. Ao final, deve-se escrever o resultado.

In [ ]:
n = int(input("Digite um numero: "))

for i in range(1, n + 1):
  print(i)


Digite um numero: 50
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50


11. Faça um programa em Python para ler 10 valores inteiros. Para cada valor fornecido escreva uma mensagem que indica se cada valor fornecido é PAR ou ÍMPAR.

In [ ]:
for i in range(10):
  x = int(input("Digite um numero para vermos se é impar ou par: "))

  if x % 2 == 0:
    print("Par")
  else:
    print("Impar")


###12. Faça um programa em Python que leia a idade de 10 alunos, calcule e escreva a média de idade destes.

In [ ]:
soma_idades = 0

for i in range(10):
    idade = int(input(f"Digite a idade do aluno {i + 1}: "))
    soma_idades += idade

media = soma_idades / 10

print(f"\nA média de idade dos 10 alunos é: {media:.2f} anos.")

Digite a idade do aluno 1: 10
Digite a idade do aluno 2: 10
Digite a idade do aluno 3: 10
Digite a idade do aluno 4: 10
Digite a idade do aluno 5: 10
Digite a idade do aluno 6: 10
Digite a idade do aluno 7: 10
Digite a idade do aluno 8: 10
Digite a idade do aluno 9: 10
Digite a idade do aluno 10: 10

A média de idade dos 10 alunos é: 10.00 anos.


###13. Faça um algoritmo em Python que leia dois valores inteiros e escreva se eles estão em ordem crescente, decrescente ou são iguais, após isso escreva todos os valores do menor ao maior.

In [ ]:
num1 = int(input("Digite o primeiro numero: "))
num2 = int(input("Digite o segundo numero: "))

if num1 > num2:
  print("Crescente", num2, num1)
elif num1 < num2:
  print("Decrescente", num2, num1)
else:
  print("São iguais", num1,)

for i in range(num1, num2 + 1):
  print(i)



Digite o primeiro numero: 50
Digite o segundo numero: 30
Crescente 30 50


###14. Escrever um programa em Python que lê 10 valores para n, um de cada vez, todos inteiros e positivos, e para cada n lido, escreva a tabuada de 1 até n de n.

1 x n = n

2 x n = 2n

...

n x n = n2

In [ ]:
# Programa que imprime a tabuada de n até n x n, para 10 valores

for i in range(10):
    n = int(input(f"Digite o {i + 1}º valor (inteiro e positivo): "))

    if n > 0:
        print(f"Tabuada do número {n}:")
        for j in range(1, n + 1):
            print(f"{j} x {n} = {j * n}")
        print()  # Linha em branco para separar as tabuadas
    else:
        print("Valor inválido. Digite apenas números inteiros positivos.\n")

Digite o 1º valor (inteiro e positivo): 1
Tabuada do número 1:
1 x 1 = 1

Digite o 2º valor (inteiro e positivo): 2
Tabuada do número 2:
1 x 2 = 2
2 x 2 = 4

Digite o 3º valor (inteiro e positivo): 3
Tabuada do número 3:
1 x 3 = 3
2 x 3 = 6
3 x 3 = 9

Digite o 4º valor (inteiro e positivo): 5
Tabuada do número 5:
1 x 5 = 5
2 x 5 = 10
3 x 5 = 15
4 x 5 = 20
5 x 5 = 25

Digite o 5º valor (inteiro e positivo): 4
Tabuada do número 4:
1 x 4 = 4
2 x 4 = 8
3 x 4 = 12
4 x 4 = 16

Digite o 6º valor (inteiro e positivo): 6
Tabuada do número 6:
1 x 6 = 6
2 x 6 = 12
3 x 6 = 18
4 x 6 = 24
5 x 6 = 30
6 x 6 = 36

Digite o 7º valor (inteiro e positivo): 10
Tabuada do número 10:
1 x 10 = 10
2 x 10 = 20
3 x 10 = 30
4 x 10 = 40
5 x 10 = 50
6 x 10 = 60
7 x 10 = 70
8 x 10 = 80
9 x 10 = 90
10 x 10 = 100

Digite o 8º valor (inteiro e positivo): 1
Tabuada do número 1:
1 x 1 = 1

Digite o 9º valor (inteiro e positivo): 1
Tabuada do número 1:
1 x 1 = 1

Digite o 10º valor (inteiro e positivo): 1
Tabuada do númer

1) Escreva um programa que lê um número inteiro N do usuário. O programa deverá escrever na tela o fatorial desse número (N!).

Exemplo de entrada:

5

Saída:

120

Exemplo de entrada:

8

Saída:

40320

In [ ]:
n = int(input("Digite um numero:"))

fatorial = 1
for i in range(1, n + 1):
  fatorial *= i

print(f"O numero de {n} é {fatorial}")

Digite um numero:8
O numero de 8 é 40320


## 2) Escreva um programa para uma urna eletrônica. O programa fica lendo os votos até que o usuário informe o valor 4444444 (código de finalização da urna). Após a finalização da urna, o programa deverá escrever na tela a quantidade de votos e o percentual de votos para cada uma das opções abaixo:
a. Votos no Candidato 77

b. Votos no Candidato 39

c. Votos no Candidato 66

d. Voto em branco (0)

e. Votos nulos (nenhuma das opções anteriores)

f. Votos válidos (aqueles computados para algum candidato)

g. Votos não válidos

In [ ]:
def main():
    votos_77 = 0
    votos_39 = 0
    votos_66 = 0
    votos_branco = 0
    votos_nulos = 0

    print("Urna Eletrônica - Digite o número do seu voto:")
    print("77 - Candidato 77")
    print("39 - Candidato 39")
    print("66 - Candidato 66")
    print("0  - Voto em Branco")
    print("Digite 4444444 para finalizar a votação.")

    while True:
        try:
            voto = int(input("Digite seu voto: "))

            if voto == 4444444:
                break
            elif voto == 77:
                votos_77 += 1
            elif voto == 39:
                votos_39 += 1
            elif voto == 66:
                votos_66 += 1
            elif voto == 0:
                votos_branco += 1
            else:
                votos_nulos += 1
        except ValueError:
            print("Entrada inválida. Digite um número.")

    total_votos = votos_77 + votos_39 + votos_66 + votos_branco + votos_nulos
    votos_validos = votos_77 + votos_39 + votos_66
    votos_nao_validos = votos_branco + votos_nulos

    def percentual(valor):
        return (valor / total_votos * 100) if total_votos > 0 else 0

    print("\nResultado da Votação:")
    print(f"Total de votos: {total_votos}")
    print(f"Votos no Candidato 77: {votos_77} ({percentual(votos_77):.2f}%)")
    print(f"Votos no Candidato 39: {votos_39} ({percentual(votos_39):.2f}%)")
    print(f"Votos no Candidato 66: {votos_66} ({percentual(votos_66):.2f}%)")
    print(f"Votos em branco: {votos_branco} ({percentual(votos_branco):.2f}%)")
    print(f"Votos nulos: {votos_nulos} ({percentual(votos_nulos):.2f}%)")
    print(f"Votos válidos: {votos_validos} ({percentual(votos_validos):.2f}%)")
    print(f"Votos não válidos: {votos_nao_validos} ({percentual(votos_nao_validos):.2f}%)")

if __name__ == "__main__":
    main()

Urna Eletrônica - Digite o número do seu voto:
77 - Candidato 77
39 - Candidato 39
66 - Candidato 66
0  - Voto em Branco
Digite 4444444 para finalizar a votação.
Digite seu voto: 77
Digite seu voto: 39
Digite seu voto: 77
Digite seu voto: 77
Digite seu voto: 66
Digite seu voto: 39
Digite seu voto: 4444444

Resultado da Votação:
Total de votos: 6
Votos no Candidato 77: 3 (50.00%)
Votos no Candidato 39: 2 (33.33%)
Votos no Candidato 66: 1 (16.67%)
Votos em branco: 0 (0.00%)
Votos nulos: 0 (0.00%)
Votos válidos: 6 (100.00%)
Votos não válidos: 0 (0.00%)
